outlines/generate/choice.py

```diff
@@ -12,12 +15,19 @@
 
 @singledispatch
 def choice(
-    model, choices: List[str], sampler: Sampler = multinomial()
+    model, choices: Union[List[str], type[Enum]], sampler: Sampler = multinomial()
 ) -> SequenceGeneratorAdapter:
-    regex_str = r"(" + r"|".join(choices) + r")"
+    if isinstance(choices, type(Enum)):
+        regex_str = build_regex_from_schema(pyjson.dumps(get_schema_from_enum(choices)))
+    else:
+        choices = [re.escape(choice) for choice in choices]  # type: ignore
+        regex_str = r"(" + r"|".join(choices) + r")"
 
     generator = regex(model, regex_str, sampler)
-    generator.format_sequence = lambda x: x
+    if isinstance(choices, type(Enum)):
+        generator.format_sequence = lambda x: pyjson.loads(x)
```

Comment:
Hey @rlouf 🙋‍♂️ 
I have a design related question: in case we pass an enum to `choice`, are we expecting to return an instance of that enum or its value? If the latter we are fine, if the former we need to reconstruct the enum in here
Thx 🙏

Reply:
Not sure what you mean but returning N instance, but we definitely just want the value.

Reply:
We are good for review then